scripts/validation/validate_all_environments.py

```
@@ -0,0 +1,839 @@
+                "Minimum 10 unique characters required. "
+                "Generate with: openssl rand -hex 32"
+            )
+
+        return v
+
+    @field_validator("OPENAI_API_KEY")
+    @classmethod
+    def validate_openai_key(cls, v: str | None) -> str | None:
+        """Validate OpenAI API key format"""
+        if v and not v.startswith("sk-"):
+            raise ValueError("OpenAI API key must start with 'sk-'")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)